source/code/plugin/health/health_missing_signal_generator.rb

```diff
@@ -0,0 +1,142 @@
+                            {"timestamp" => Time.now.utc.iso8601, "state" => HealthMonitorStates::UNKNOWN, "details" => ""}
+                        )
+                        if !node_name.nil? && nodes.include?(node_name)
+                            new_monitor.state = HealthMonitorStates::UNKNOWN
+                            new_monitor.details["state"] = HealthMonitorStates::UNKNOWN
+                            new_monitor.details["details"] = "Node present in inventory but no signal for #{monitor.monitor_id} from node #{node_name}"
+                            @unknown_signals_hash[monitor_instance_id] = new_monitor
+                        elsif !node_name.nil? && !nodes.include?(node_name)
+                            new_monitor.state = HealthMonitorStates::NONE
+                            new_monitor.details["state"] = HealthMonitorStates::NONE
+                            new_monitor.details["details"] = "Node NOT present in inventory.  node:  #{node_name}"
+                        end
+                        missing_signals_map[monitor_instance_id] = new_monitor
+                        log.info "Added missing signal #{new_monitor.monitor_instance_id} #{new_monitor.state}"
+                    elsif HealthMonitorHelpers.is_pods_ready_monitor(monitor.monitor_id)
+                        lookup = "#{monitor.labels['container.azm.ms/namespace']}~~#{monitor.labels['container.azm.ms/workload-name']}"
+                        new_monitor = HealthMonitorRecord.new(
+                            monitor.monitor_id,
+                            monitor.monitor_instance_id,
+                            Time.now.utc.iso8601,
+                            monitor.state,
+                            monitor.labels,
+                            monitor.config,
+                            {"timestamp" => Time.now.utc.iso8601, "state" => HealthMonitorStates::UNKNOWN, "details" => ""}
+                        )
+                        if !lookup.nil? && workload_names.include?(lookup)
+                            new_monitor.state = HealthMonitorStates::UNKNOWN
+                            new_monitor.details["state"] = HealthMonitorStates::UNKNOWN
+                            new_monitor.details["details"] = "Workload present in inventory. But no signal for #{lookup}"
+                            @unknown_signals_hash[monitor_instance_id] = new_monitor
+                        elsif !lookup.nil? && !workload_names.include?(lookup)
+                            new_monitor.state = HealthMonitorStates::NONE
+                            new_monitor.details["state"] = HealthMonitorStates::NONE
+                            new_monitor.details["details"] = "Workload #{lookup} NOT present in inventory"
+                        end
+                        missing_signals_map[monitor_instance_id] = new_monitor
+                    end
+                end
+            }
+
+
+            health_monitor_records.each{|health_monitor_record|
```

Comment:
not sure how large health_monitor_records would be in case of a big cluster withg lots of nodes and each with huge number ofpods (mainly AKS-engine)

Reply:
4x nodes
2xworkloads
1xnodepools
1xnamespaces
+15 (kube-system artifacts)
is the total number of records per cluster